### PR TITLE
2 change contact area calculation of threerollpasses

### LINKED ISSUE #2: change contact area calculation of ThreeRollPasses
Similar to pyroll-project/pyroll-core#238 the calculation of zouhar contact areas should also be changed.

### PR DESCRIPTION
**Pull request by RichardPfr**
Changed the Contact area calculation so that its more representative for three-roll rolling.

Currently there is also a factor 0.999 necessary to correctly calculate the in_profile local_width - probably due to small numerical deviations.

### CODE REVIEW

**Comment by ChRen95** [comment]
Without further review of the code, the proposed tests still yield errors. 
The test that fails to converge are: 
 test.test_contact_area.test_plotted_contact_area_3fold

**Comment by RichardPfr** [comment]
Sry didnt mention:
To have these changes on the zouhar package work the changes made in pyroll-project/pyroll-core#241 are also required, as the updated contact_lines property is required. 
In my venv i included those changes with a .whl.

I suppose thats where the error comes from, but if not i can have a closer look again.

File: pyroll/zouhar_contact.py [hunk 1]
```diff
@@ -70,11 +75,16 @@ def contact_area(self: RollPass.Roll):
             and
             rp.zouhar_contact_in_width is not None
     ):
+        if "3fold" in rp.classifiers:
```
Comment (axtimhaus) [inline pyroll/zouhar_contact.py]: Maybe you can avoid such checks if you would define a new hook function for TRP only. Refactor the main logic to a simple function which is called from the respective hook impl. Then you would not need the core changes (removal of TRP default hook impls there).

**Comment by RichardPfr** [comment]
Added the ThreeRollPass handling to a separate Hook function - but i dont know what you specifically mean with not needing the core changes now? (or is this already clarified with the changes i did there?)

**Comment by ChRen95** [comment]
Did you resolve everything?

> Added the ThreeRollPass handling to a separate Hook function - but i dont know what you specifically mean with not needing the core changes now? (or is this already clarified with the changes i did there?)

**Review by axtimhaus — CHANGES_REQUESTED** [review]
This code is highly duplicated. Refactor it to a function that is called by the hooks

**Comment by axtimhaus** [comment]
> Added the ThreeRollPass handling to a separate Hook function - but i dont know what you specifically mean with not needing the core changes now? (or is this already clarified with the changes i did there?)

With this two hooks you circumvent the problem with the precedence of the hooks in the core for 3RP

**Review by ChRen95 — APPROVED** [review]

**Comment by ChRen95** [comment]
I think also the comments by @axtimhaus are resolved. So please proceed.

**Pull request merged**